flink-cdc-composer/src/main/java/org/apache/flink/cdc/composer/flink/deployment/YarnSessionDeploymentExecutor.java

```diff
@@ -0,0 +1,164 @@
+import org.apache.flink.yarn.configuration.YarnLogConfigUtil;
+
+import org.apache.flink.shaded.guava31.com.google.common.base.Joiner;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.hadoop.yarn.api.records.ApplicationId;
+import org.apache.hadoop.yarn.api.records.FinalApplicationStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/** Deploy flink cdc job by yarn session mode. */
+public class YarnSessionDeploymentExecutor extends AbstractDeploymentExecutor {
+    private static final Logger LOG = LoggerFactory.getLogger(YarnSessionDeploymentExecutor.class);
+
+    @Override
+    public PipelineExecution.ExecutionInfo deploy(
+            CommandLine commandLine,
+            Configuration flinkConfig,
+            List<Path> additionalJars,
+            Path flinkHome)
+            throws Exception {
+        LOG.info("Submitting flink job in 'Flink Yarn Session Mode'.");
+        if (flinkConfig.get(PipelineOptions.JARS) == null) {
+            flinkConfig.set(
+                    PipelineOptions.JARS, Collections.singletonList(getFlinkCDCDistJarFromEnv()));
+        }
+        flinkConfig.set(
+                YarnConfigOptions.SHIP_FILES,
+                additionalJars.stream().map(Path::toString).collect(Collectors.toList()));
+
+        ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+
+        Path pipelinePath = new Path(commandLine.getArgList().get(0));
+        FileSystem fileSystem = FileSystem.get(pipelinePath.toUri());
+        FSDataInputStream pipelineInStream = fileSystem.open(pipelinePath);
+
+        flinkConfig.set(
+                ApplicationConfiguration.APPLICATION_ARGS,
+                Collections.singletonList(mapper.readTree(pipelineInStream).toString()));
+        YarnLogConfigUtil.setLogConfigFileInConfig(
+                flinkConfig, Joiner.on(File.separator).join(flinkHome, "conf"));
+
+        flinkConfig.set(ApplicationConfiguration.APPLICATION_MAIN_CLASS, getCDCMainClass());
+        final YarnClusterClientFactory yarnClusterClientFactory = new YarnClusterClientFactory();
+        final YarnClusterDescriptor descriptor =
+                yarnClusterClientFactory.createClusterDescriptor(flinkConfig);
+        ClusterSpecification specification =
+                yarnClusterClientFactory.getClusterSpecification(flinkConfig);
+
+        ClusterClient<ApplicationId> client = null;
+        try {
+            // If applicationId is passed, we get the state of yarn; if not, we create a session
+            // cluster.
+            String applicationId = flinkConfig.get(YarnConfigOptions.APPLICATION_ID);
+            if (applicationId != null) {
```

Comment:
It seems that we cannot determine whether the application_id does not exist or is a misspelled application_id. If it is a misspelled, a new session cluster will be generated each time it is provided

Reply:
Maybe we can add a simple validate method.